pkg/hierarchy.go

```diff
@@ -16,12 +18,13 @@ type Config struct {
 }
 
 type Item struct {
-	Kind      string       `yaml:"kind" json:"kind"`
-	Name      string       `yaml:"name" json:"name"`
-	Namespace string       `yaml:"namespace,omitempty" json:"namespace,omitempty"`
-	Type      ResourceType `yaml:"type,omitempty" json:"type,omitempty"`
-	Key       string       `yaml:"key,omitempty" json:"key,omitempty"`
-	Index     int          `yaml:"index,omitempty" json:"index,omitempty"`
+	Kind          string
+	Name          string
+	Namespace     string
+	Type          ResourceType
+	Key           string
+	Index         int
+	HierarchyName string `yaml:"hierarchyName"`
```

Comment:
Also since this struct is also used by operator can you have `omitempty` here so it doesn't become a required field in the CRD